coordinator/src/main/scala/com/socrata/datacoordinator/service/Main.scala

```diff
@@ -506,7 +506,7 @@ class Main(common: SoQLCommon, serviceConfig: ServiceConfig) {
     val preferredSecondaries = secondariesLike match {
       case Some(datasetInternalName) =>
         val candidateSecondariesInGroup = secondaryGroup.instances.filter(_._2.acceptingNewDatasets).keySet
-        val secs = collocationProvider.collocator.secondariesOfDataset(datasetInternalName)
+        val secs = collocationProvider.collocator.secondariesOfDataset(datasetInternalName).filter(secondaryGroup.instances.keySet)
```

Comment:
Should we use `candidateSecondariesInGroup` since that does not contain secondaries that aren't accepting new datasets?

Reply:
Oh nm. I see that we're comparing the two below and generating an error message.

Reply:
No, this needs to include secondaries that aren't accepting new datasets.

Reply:
(because if we exclude those we'll never find a secondary that is genuinely not accepting new datasets on the next line)